src/spec.md

```diff
@@ -0,0 +1,331 @@
+   - `rpc` (string): RPC for any evm network.
+   - `ethStorageRpc` (string): The EthStorage network rpc corresponding to this evm network, the data is obtained from the EthStorage network.
+   - `privateKey` (string): Wallet private key.
+
+**Example**
+```javascript
+const config = {
+   rpc: "your_rpc",
+   ethStorageRpc: "ethstorage_rpc",
+   privateKey: "your_private_key"
+};
+
+const ethStorage = new EthStorage(config);
+```
+
+
+
+### Methods
+
+#### estimateCost
+
+**Description**: Estimate the cost of uploading data.
+
+**Parameters**
+- `key` (string): The key for the data to be written.
+- `data` (Buffer): The data to be written, its size cannot exceed the maximum value of the content that can be transferred by a blob.
+
+**Returns**
+- `cost` (Promise<Bigint>): A Promise that resolves to the cost.
+
+**Example**
+```javascript
+const cost = await ethStorage.estimateCost("dataKey", Buffer.from("some data"));
+```
+
+
+
+#### write
+
+**Description**: Asynchronously writes data to the EthStorage network.
+
+**Parameters**
+- `key` (string): The key for the data to be written.
+- `data` (Buffer): The data to be written, its size cannot exceed the maximum value of the content that can be transferred by a blob.
+
+**Returns**
+- `status` (Promise<boolean>): A Promise that resolves to the execution result. `true|false`
+
+**Example**
+```javascript
+const status = await ethStorage.write("dataKey", Buffer.from("some data"));
+```
+
+
+
+#### read
+
+**Description**: Read data asynchronously from the EthStorage network through key.
+
+**Parameters**
+- `key` (string): The key for the data to be read.
+
+**Returns**
+- `data` (Promise<Buffer>): A Promise that resolves to the content.
+
+**Example**
+```javascript
+const data = await ethStorage.read("example.txt");
+```
+
+
+#### putBlobs
+**Description**: Batch upload blob data to the EthStorage network.
+
+**Parameters**
+- `number` (number): Number of blobs.
+- `data` (Buffer): Blob content to be written, its size cannot exceed the maximum value of the content that can be transferred by a blob.
+
+**Returns**
+- `status` (Promise<boolean>): A Promise that resolves to the execution result. `true|false`
+
+**Example**
+```javascript
+const blobData = Buffer.from("some data");
+const status = await ethStorage.putBlobs(number, blobData);
+```
+
+
+
+<p id="FlatDirectory"></p>
+
+## 4. FlatDirectory Class
+
+### Constructor
+
+**Description**: Create an instance of `FlatDirectory`.
+
+**Parameters**
+- `config` (object): Configuration object containing necessary settings.
+    - `rpc` (string): RPC for any evm network.
+    - `ethStorageRpc` (string): The EthStorage network rpc corresponding to this evm network, the data is obtained from the EthStorage network.
+    - `privateKey` (string): Wallet private key.
+    - `address` (string, optional): FlatDirectory contract address. If it does not exist, the `deploy` method can be called to create one.
```

Comment:
If it does not exist, the `deploy` method need to be called before calling other function?

Reply:
If the answer is yes, I would suggest that moving the explanation of the deploy function to the first one.